test/extended/util/cli.go

```
@@ -0,0 +1,219 @@
+	}
+	c.cmd.SetOutput(c.stdout)
+	return c
+}
+
+// Run executes given OpenShift CLI command verb (iow. "oc <verb>").
+// This function also override the default 'stdout' to redirect all output
+// to a buffer and prepare the global flags such as namespace and config path.
+func (c *CLI) Run(verb string) *CLI {
+	out := new(bytes.Buffer)
+	nc := &CLI{
+		verb:          verb,
+		kubeFramework: c.KubeFramework(),
+		cmd:           cli.NewCommandCLI("oc", "openshift", out),
+		globalArgs: []string{
+			verb,
+			fmt.Sprintf("--namespace=%s", c.Namespace()),
+			fmt.Sprintf("--config=%s", c.configPath),
+		},
+	}
+	return nc.SetOutput(out)
+}
+
+// Template sets a Go template for the OpenShift CLI command.
+// This is equivalent of running "oc get foo -o template -t '{{ .spec }}'"
+func (c *CLI) Template(t string) *CLI {
+	if c.verb != "get" {
+		FatalErr("Cannot use Template() for non-get verbs.")
+	}
+	templateArgs := []string{"--output=template", fmt.Sprintf("--template=%s", t)}
+	commandArgs := append(c.commandArgs, templateArgs...)
+	c.finalArgs = append(c.globalArgs, commandArgs...)
+	c.cmd.SetArgs(c.finalArgs)
+	return c
+}
+
+// Args sets the additional arguments for the OpenShift CLI command
+func (c *CLI) Args(args ...string) *CLI {
+	c.commandArgs = args
+	c.finalArgs = append(c.globalArgs, c.commandArgs...)
+	c.cmd.SetArgs(c.finalArgs)
+	return c
+}
+
+func (c *CLI) printCmd() string {
+	return strings.Join(c.finalArgs, " ")
+}
+
+// Output executes the command and return the output as string
+func (c *CLI) Output() (string, error) {
+	if c.verbose {
+		fmt.Printf("DEBUG: oc %s\n", c.printCmd())
+	}
+	err := c.cmd.Execute()
+	if err != nil {
+		FatalErr(err)
+	}
+	out := c.stdout.(*bytes.Buffer)
+
+	return strings.TrimSpace(out.String()), err
+}
+
+// Execute executes the current command and return error if the execution failed
+// This function will set the default output to stdout.
+func (c *CLI) Execute() error {
+	out, err := c.Output()
```

This first err is ignored.